src/components/pt-element/PtElementInfo.tsx

```diff
@@ -1,41 +1,37 @@
-import autobind from "autobind-decorator";
-import classNames from "classnames";
 import * as React from "react";
+import classNames from "classnames";
+
 import { IElement } from "../../Element";
 import { getElementLocales } from "../../ElementManager";
 import Button from "../shared/button/Button";
 import "./PtElement.scss";
 
-export interface IPtElementInfoProps {
+export interface PtElementInfoProps {
   element: IElement;
   onClick?: (element: IElement) => void;
 }
 
-@autobind
-class PtElementInfo extends React.Component<IPtElementInfoProps> {
-  public render() {
-    const { element } = this.props;
-    const elementLocales = getElementLocales(element);
-
-    return (
-      <Button
-        onClick={this.onClick}
-        className={classNames("pt-element", "element", element.group)}
-      >
-        <div className="pt-element__atomic">{element.atomic}</div>
-
-        <div className="pt-element__symbol">{element.symbol}</div>
-
-        <div className="pt-element__name">{elementLocales.name}</div>
-      </Button>
-    );
-  }
-
-  private onClick() {
-    if (this.props.onClick) {
-      this.props.onClick(this.props.element);
+function PtElementInfo({ element, onClick }: PtElementInfoProps) {
+  const elementLocales = getElementLocales(element);
+
+  const onElementButtonClick = () => {
+    if (onClick) {
+      onClick(element);
     }
```

Comment:
just an FYI, you can simplify this in TS with optional chaining:

```suggestion
    onClick?.(element);
```

Reply:
cool I didn't know that we can do that with functions 🚀